### DISCUSSION TITLE
Ability to rename DAG

### DISCUSSION THREAD

**Discussion by sann05** [apache/airflow#28987] [General]
### Description

I know that the DAG's id is added in the python script, but I have had this intention so many times.

I have to have the ability to change DAG's id without losing all history. 

Let's say it should work like that:
Whenever you change `dag_id` in the dag property, you'll have to add `prev_dag_id` if you want it will be renamed
If the `prev_dag_id` property is added, but there is no dag with such `dag_id`, no error is raised. A new dag has to be created

### Use case/motivation

I am a person who likes to keep their DAGs well-named and organized.
But sometimes I create some DAG thinking that it will rely upon some data source, say "DOWNLOAD.data_source.entity1".
But then, after I added 20 other DAGs, I realized that the best name for a DAG would be "DOWNLOAD.data_source_type.entity1".

You would say that It's better to call your dags like "DOWNLOAD.entity1" and add tags, so I can change tags whenever I want. But what If I have 5 of the same entities I download from different data sources?

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by potiuk** [comment]
This is part of DAG versioning bigger feature and until we work out the whole big improvement for that, It's highly unlikely we will implement just rename of DAG: https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-36+DAG+Versioning 

Thera are far too many consequences of renaming DAG to make it into just a "feature" - it fundamentally changes many assumptions under the hood and complexity of it is far bigger than you might think.

The AIP is not even yet approved, so do not expect anything soon (unless you would like to lead the efffort, propose AIP, get it discussed and approved. 

Converting it into a discussion as this is not likely to be ever implemented before versioning is addressed as a whole.
